operator/internal/resources/v2_node_pools.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package resources
+
+import (
+	"context"
+
+	"github.com/redpanda-data/redpanda-operator/charts/redpanda"
+	redpandav1alpha2 "github.com/redpanda-data/redpanda-operator/operator/api/redpanda/v1alpha2"
+	"github.com/redpanda-data/redpanda-operator/pkg/gotohelm/helmette"
+	"github.com/redpanda-data/redpanda-operator/pkg/kube"
+	appsv1 "k8s.io/api/apps/v1"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// V2NodePoolRenderer represents a node pool renderer for v2 clusters.
+type V2NodePoolRenderer struct {
+	kubeConfig clientcmdapi.Config
```

Comment:
This will start to break soon, sorry :/ I changed gotohelm to use rest.Configs to support vClusters in tests.